app/utils/calculation/calculation.ts

```diff
@@ -136,11 +137,63 @@ const getPeriodicityText = ({
   return periodicityText;
 };
 
+const getImperialMetricValue = (
+  metricValue: number,
+  useMetricUnits: boolean,
+  measureType: MeasureType
+): number => {
+  if (useMetricUnits) {
+    return metricValue;
+  } else {
+    if (measureType === MeasureType.mass) {
+      /* kg -> lbs */
+      return metricValue * 2.205;
+    } else if (measureType === MeasureType.length) {
+      /* km -> miles
+      note: deviates from NMF.earth's standard of using meters as a reference */
+      return metricValue / 1.609;
+    }
+  }
+};
+
+const getDisplayUnitsValue = (kgValue: number, useMetricUnits: boolean): number => {
+  if (kgValue <= 1 && useMetricUnits) {
+    return kgValue * 1000;
+  } else if (kgValue > 1 && kgValue <= 1000 && useMetricUnits) {
+    return kgValue;
+  } else if (kgValue > 1000 && useMetricUnits) {
+    return kgValue / 1000;
+  } else if (kgValue <= 0.454 && !useMetricUnits) {
+    return getImperialMetricValue(kgValue, useMetricUnits, MeasureType.mass) * 16;
+  } else {
+    /*if (kgValue > 0.454 && !useMetricUnits)*/
+    return getImperialMetricValue(kgValue, useMetricUnits, MeasureType.mass);
+  }
+};
+
+const getDisplayUnits = (kgValue: number, useMetricUnits: boolean, useSymbol = true): string => {
```

Comment:
This can be simplified / more readable : 
```
const getDisplayUnits = (kgValue: number, useMetricUnits: boolean, useSymbol = true): string => {
  const suffix = useSymbol ? "_SYMBOL" : "_FULL";

  if (useMetricUnits) {
    if (kgValue <= 1) {
      return t(`GRAMS${suffix}`);
    } else if (kgValue > 1 && kgValue <= 1000) {
      return t(`KILOGRAMS${suffix}`);
    } else {
      return t(`TONNES${suffix}`);
    }
  } else {
    if (kgValue <= 0.454) {
      return t(`OUNCES${suffix}`);
    } else {
      return t(`POUNDS${suffix}`);
    }
  }
};
```

Reply:
Yeah that's much more readable. Satisfying too. Thanks for the coaching. I like to code, but I know I write spaghetti haha, so learning patterns like this (extracting booleans common throughout an if-else block to an outer if-else block; not sure if there's a technical name for it) is really cool and one of the reasons I wanted to start contributing to open source. It makes sense (maybe even common sense to others), it just didn't occur to me that doing so would make the code easier to read and understand. Taking advantage of the naming "_SYMBOL/_FULL" convention to simplify all the ternaries to a single ternary is really satisfying too. Would you be able to recommend any resources for learning practices like these?

Reply:
I would say that you usually don't need any comments when the code is "okayish" and the naming of the variables/function are also important, I prefer long and explicit, but that could be arguable sometimes. For e.g, it's usually good practice to have `has`/`is` prefix for `bool` variables and then when you see duplicate code it means you can refactor. I would say it comes with times and looking at code from different large projects (with several contributors) with different way of coding, it helps you to figure out your style, and teach how to write reusable and easy to understand code. 
Also, for this scenario, I was also thinking of using functional programming (FP) with ramda.js, but it's not so necessary as it's not a big piece of code. However, when you have to process and transform a lot of data, FP helps you to write concise & reusable code (for e.g have a look to `getEmissions` in `EmissionsScreen.selectors.ts`)

Reply:
Hmm okay. I have noticed that the long and explicit names in this repo have made it pretty easy to follow without extensive commenting. Duplicate code => you can refactor; of course. I mean feels like that should've been common sense again, and it's something I've sometimes followed naturally, but I'm not sure I've ever codified it in my head. 
And sweet, thanks for formally introducing me to FP. Just did some research on it. Looks like another thing I've been _kind of_ using but not with much intention. That flow in the Emissions selector is very readable and concise. I hope I come across some kind of transformation-heavy situation soon to apply it